src/ASM/LR/ASMu3D.C

```diff
@@ -1962,21 +1950,26 @@ std::vector<int> ASMu3D::getFaceNodes (int face, int basis, int orient) const
 }
 
 
-void ASMu3D::getBoundaryNodes (int lIndex, IntVec& nodes, int basis, int, bool) const
+void ASMu3D::getBoundaryNodes (int lIndex, IntVec& nodes, int basis,
+                               int, int orient, bool local) const
 {
   if (basis == 0)
     basis = 1;
 
   if (!this->getBasis(basis)) return; // silently ignore empty patches
 
-  nodes = this->getFaceNodes(lIndex, basis);
+  nodes = this->getFaceNodes(lIndex, basis, orient);
 
 #if SP_DEBUG > 1
   std::cout <<"Boundary nodes in patch "<< idx+1 <<" edge "<< lIndex <<":";
   for (size_t i = 0; i < nodes.size(); i++)
     std::cout <<" "<< nodes[i];
   std::cout << std::endl;
 #endif
+
+  if (!local)
+    for (auto& node : nodes)
```

Comment:
`auto` is convenient, but when the actual type is simple, as in this case, I prefer to use that instead, for readability. I.e., in this case `for (int& node : nodes)`